app/models/pipeline_run.rb

```diff
@@ -335,6 +335,12 @@ def load_taxons(downloaded_json_path, refined = false)
     pipeline_output_dict = json_dict['pipeline_output']
     pipeline_output_dict.slice!('taxon_counts_attributes')
 
+    # check if there's any record loaded into taxon_counts. If so, skip
+    check_count_type = refined ? 'NT+' : 'NT'
+    loaded_records = TaxonCount.where(pipeline_run_id: id)
+                               .where(count_type: check_count_type).count
+    return if loaded_records > 0
```

Comment:
is this the right thing to do? do all the records get loaded in one transaction? or is it possible that some taxons are loaded but not all of them?

Reply:
they are loaded in one transaction because how update_attributes work